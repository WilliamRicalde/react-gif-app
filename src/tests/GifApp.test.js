import { shallow } from "enzyme"
import GifApp from "../GifApp"

describe('Pruebas en el Gifapp', () => {
    test('debe hacer match con el snapshot', () => {
        const wrapper = shallow(<GifApp />);

        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar dos elementos', () => {
        const categories = ['One Punch', 'Dragon Ball']
        const wrapper = shallow( <GifApp defaultCategories={categories} />)

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
    })
    
    
})
