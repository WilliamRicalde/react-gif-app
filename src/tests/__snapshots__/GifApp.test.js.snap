// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el Gifapp debe de mostrar dos elementos 1`] = `
<Fragment>
  <h2>
    Giphy API App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;

exports[`Pruebas en el Gifapp debe hacer match con el snapshot 1`] = `
<Fragment>
  <h2>
    Giphy API App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
